devito/finite_differences/equation.py

```diff
@@ -78,7 +78,8 @@ def subdomain(self):
     def evaluate(self):
         """
         Evaluate the Equation or system of Equations.
-        The rhs of the Equation is evaluated at the indices of the lhs if required.
+
```

Comment:
?

And really not sure should be in FD

Reply:
I can move it to somewhere else if you prefer -- types ?

definitely it needs to get out of the root directory

I think `finite_difference` is okish to be honest... an alternative is `types`. I can move it there if you prefer?

Reply:
I'd say in `extended_sympy` personally

Reply:
no, definitely it can't go there. 

`Eq` is an actual type of the DSL, and `extended_sympy` has got nothing to do with it, it's just some sympy refinements.

The DSL types are spread across `finite_differences`, `types`, and `operator`. This separation is quite neat I think, so I'm not debating it. But I agree with you `Eq` can have a better place, only not `extended_sympy`. I'll think about that, but `types` actually might not be such a bad place. Or peraphs the new `operator` directory too... but it's a bit weird, because I'd prefer to keep `Eq` together with the other symbolic objects

Reply:
I get your point. I think type is a bit too overcrowded though, maybe we should split it into
- "datatypes" (Function, ...)
-"devito-sympy types" (Eq, ..)

Reply:
Maybe, I don't know, but I'd refrain from creating new directories and shuffling other modules around in this PR . If, instead, it's just a matter of moving a specific module (`equation.py`), then not a problem; however, if it's more than that (and in principle I'd have no objections) I'd wait until the stack of currently open PRs will have landed

Reply:
I think I prefer it in types then.

Reply:
OK, will do